### DISCUSSION TITLE
Do we need `~/.terminfo` symlink on Linux?

### DISCUSSION THREAD

**Discussion by salmankhilji** [kovidgoyal/kitty#5923] [Q&A] [answered]
I am questioning the need to continue to create the following symlink in my kitty installation script:

```
ln -rfs ~/.local/kitty.app/lib/kitty/terminfo ~/.terminfo
```
(Intended to run on a Linux system.)

[This](https://sw.kovidgoyal.net/kitty/faq/#keys-such-as-arrow-keys-backspace-delete-home-end-etc-do-not-work-when-using-su-or-sudo) FAQ entry mentions its need with `su` or `sudo`.  However,  arrow keys work just fine in `vim` on my machine while running with `sudo` _without_ this symlink.

Is there any other program that relies on this folder other than kitty itself?

Yes, I do have `$TERMINFO` (magically defined by `kitty`, I assume):

  ```
  $ printenv TERMINFO
  /home/precor/.local/kitty.app/lib/kitty/terminfo
  ```

I must have followed advice from an older FAQ to create this symlink, but somehow can not justify its existence anymore.

### Use Case

The reason I ask is that I do not want to update this symlink when switching between release/nightly and build-from-source versions.

The FAQ entry mentions doing something like this:

  ```
  sudo … TERMINFO=$HOME/.terminfo bash -i
  ```

Wouldn't it be more correct if we rather do this:

```
sudo … TERMINFO=$KITTY_INSTALLATION_DIR/terminfo bash -i
```

The latter uses `$KITTY_INSTALLATION_DIR`, which would automatically point to the correct kitty version no matter its a binary install or built from source.

**Comment by kovidgoyal** [comment, marked as answer]
On Sun, Jan 22, 2023 at 11:02:01PM -0800, salmankhilji wrote:
> I am questioning the need to continue to create the following symlink in my kitty installation script:
> 
> ```
> ln -rfs ~/.local/kitty.app/lib/kitty/terminfo ~/.terminfo
> ```
> (Intended to run on a Linux system.)
> 
> [This](https://sw.kovidgoyal.net/kitty/faq/#keys-such-as-arrow-keys-backspace-delete-home-end-etc-do-not-work-when-using-su-or-sudo) FAQ entry mentions its need with `su` or `sudo`.  However,  arrow keys work just fine in `vim` on my machine while running with `sudo` _without_ this symlink.
> 
> Is there any other program that relies on this folder other than kitty itself?

kitty does not rely on it, programs that use ncurses or a similar
library that reads terminfo do rely on it. vim
relies on it as well, though the symptoms of it being missing are
subtle there and highly dependent on vim version.

> 
> Yes, I do have `$TERMINFO` (magically defined by `kitty`, I assume):
> 
>   ```
>   $ printenv TERMINFO
>   /home/precor/.local/kitty.app/lib/kitty/terminfo
>   ```
> 
> I must have followed advice from an older FAQ to create this symlink, but somehow can not justify its existence anymore.
> 
> ### Use Case
> 
> The reason I ask is that I do not want to update this symlink when switching between release/nightly and build-from-source versions.

You dont need the symlink ever.

> 
> The FAQ entry mentions doing something like this:
> 
>   ```
>   sudo … TERMINFO=$HOME/.terminfo bash -i
>   ```
> 
> Wouldn't it be more correct if we rather do this:
> 
> ```
> sudo … TERMINFO=$KITTY_INSTALLATION_DIR/terminfo bash -i

That will only work on the local machine not when sshed into a remote
server. The ssh kitten automatically adds the kitty terminfo to
~/.terminfo when connecting to a remote server.
